### DISCUSSION TITLE
Class B operation

### DISCUSSION THREAD

**Discussion by tsuijten** [Lora-net/LoRaMac-node#1540] [Q&A]
In my specific use case `Class B` would be the perfect solution with a good balance between responsiveness and power consumption. I want to be able to send a downlink from the gateway to the lora device to "activate" my device with a maximum time window of about 120 seconds. I could do this with `Class A` (just sending a LoRaWAN message every 2 minutes) but then I would probably be going over the fair use policies. 

However so far I've not been able to get `Class B` working.

I've enable `Class B` in Chirpstack with the correct setting to my EU868 region:
![Screenshot 2023-05-09 at 17 21 45](https://github.com/Lora-net/LoRaMac-node/assets/772711/212d545f-d7f4-4e15-9a16-659f6816e557)

I've set `LORAWAN_DEFAULT_CLASS` to `CLASS_B`

When I send the first few messages out I see `DeviceTimeReq` and the see `OnSysTimeUpdate isSynchronized: 1`

But I'm not able to schedule downlink messages with the 128 second interval window.

Could anybody please help me with getting this working?

**Comment by mluis1** [comment]
For Class B operation a gateway with a GPS module and its PPS signal connected to the SX130x chip is required.
Do you have such kind of gateways?

If yes, then usually an end-device connects to the network server and then tries to acquire the beacon signal sent by the gateways. Once this step is done the end-device is able to enable the Class B mode of operation,
It is only once the Class B mode of operation is active that you will be able to send downlinks.

I would recommend to read the LoRaWAN 1.0.4 specification as well as potentially read the following article [In-Depth: LoRaWAN® 1.0.4 Class B and C for End Devices](https://lora-developers.semtech.com/documentation/tech-papers-and-guides/lorawan-1.0.4-class-b-and-c-for-end-devices/lorawan-1.0.4-class-b-and-c)

**Reply by tsuijten** [reply]
Yeah, I was reading through that and saw that you actually need a router that support Class B. However, I'm not sure if mine does, I cannot find anything in the spec-sheet if Class B is supported. I have a [MikroTik LtAP LR8 LTE kit ](https://mikrotik.com/product/ltap_lr8_lte_kit) gateway that actually has GPS. But I'm not sure if this gateway would support Class B, is there anything you see that indicates whether Class B is supported or not?

**Reply by tsuijten** [reply]
I also tried to ask the question on the MikroTik Forum, but my post keeps being deleted without any notice 🤔

**Reply by rohinsuthar** [reply]
Yes, i also use the gateway of semtech corecell reference device so this type of gateway not come with gps module then no PPS pulse is generated and no beacon initiated
